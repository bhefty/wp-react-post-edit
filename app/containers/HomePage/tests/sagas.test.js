/**
 * TODO: Add sufficient Saga testing
 */

import { takeLatest, take } from 'redux-saga/effects'

import {
  LOAD_RECENT_POSTS,
  DELETE_POST
} from 'containers/HomePage/constants'

import {
  getRecentPosts,
  recentPosts,
  watchDeletePost
} from '../sagas'

/* eslint-disable redux-saga/yield-effects */

describe('recentPost Saga', () => {
  const recentPostsSaga = recentPosts()

  it('should start task to watch for the LOAD_TRANSLATION action', () => {
    const takeLatestDescriptor = recentPostsSaga.next().value
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_RECENT_POSTS, getRecentPosts))
  })
})

describe('watchDeletePost saga', () => {
  const watchDeletePostSaga = watchDeletePost()
  it('should start task to watch for DELETE_POST action', () => {
    const takeDescriptor = watchDeletePostSaga.next().value
    expect(takeDescriptor).toEqual(take(DELETE_POST))

    const callDescriptor = watchDeletePostSaga.next().value
    expect(callDescriptor).toMatchSnapshot()
  })
})
