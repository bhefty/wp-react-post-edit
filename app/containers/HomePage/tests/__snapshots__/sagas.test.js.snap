// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watchDeletePost saga should start task to watch for DELETE_POST action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;
